src/foundation.cs

```diff
@@ -3152,8 +3152,7 @@ public interface NSException : NSCoding, NSCopying {
 		string[] CallStackSymbols { get; }
 	}
 
-	public delegate void NSExpressionHandler (NSObject evaluatedObject, NSExpression [] expressions, NSMutableDictionary context);
-	
+	public delegate NSObject NSExpressionHandler (NSObject evaluatedObject, NSExpression [] expressions, NSMutableDictionary context);
```

Comment:
Indeed, our API is wrong. We should return a NSObject, else it makes no sense. We can add a #if with the new version.